src/hotspot/cpu/riscv/templateTable_riscv_64.cpp

```diff
@@ -2192,10 +2165,7 @@ void TemplateTable::_return(TosState state) {
   __ remove_activation(state, /* throw_monitor_exception */ true);
   // Restoration of lr done by remove_activation.
   switch (state) {
-    // Narrow result if state is itos but result type is smaller.
-    // Need to narrow in the return bytecode rather than in generate_return_entry
-    // since compiled code callers expect the result to already be narrowed.
-    case itos: __ narrow(R25_tos); /* fall through */
+    case itos:
```

Comment:
why did you remove the narrowing?

Reply:
registers are always correctly sign-extended on riscv